plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -271,10 +279,27 @@ private fun getPackageReferences(podfileLock: File): Set<PackageReference> {
             dependencies = dependenciesForName[name].orEmpty().mapTo(mutableSetOf()) { createPackageReference(it) }
         )
 
-    return root.get("DEPENDENCIES").mapTo(mutableSetOf()) { node ->
+    root.get("DEPENDENCIES").mapTo(references) { node ->
         val name = node.textValue().substringBefore(" ")
         createPackageReference(name)
     }
+
+    root.get("CHECKOUT OPTIONS")?.fields()?.asSequence()?.forEach {
+        val checkout = it.value as ObjectNode
+        val url = checkout[":git"].textValueOrEmpty()
+        val revision = checkout[":commit"].textValueOrEmpty()
+        val id = Identifier("Pod", "", it.key, versionForName[it.key].orEmpty())
+
+        externalSources[id] = Package(
+            id = id,
```

Comment:
Would you know how the version string in the lockfile is determined?
I'm asking, because if commonly many revisions are used for same version string for such unpublished artifacts, maybe we should use the SHA1 in the version part of the identifier?

Reply:
No, I don't know. That why I'm using `versionForName[it.key].orEmpty()` when determining the `id` to eventually get a "real" version in addition to the SHA1.

Reply:
> so that we do not risk that e.g. curation get filed for package versions which actually are used in combination with many different SHA1s.

Sorry, I do not understand that sentence / the risk. Could you maybe give a concrete example?

Reply:
> Sorry, I do not understand that sentence / the risk. Could you maybe give a concrete example?

I was wondering how the version string arrives at the lockfile. If the procedure starts with a specified the SHA1
and then determines the version based on

1. The package metadata found in the repo for given SHA1
2. Or the youngest Git tag preceeding the SHA1

then many SHA1's may correspond to a single ID.
That would be problematic with our package curation mechanism.
Does it become clearer with this?

Reply:
> I was wondering how the version string arrives at the lockfile.

I've tested it at the example of `MLHudAlert` by specifying a custom commit in `Podfile` that does not point to a tag and regenerating the lockfile. The version that ends up in `Podfile.lock` is whatever the project's `.podspec` file contains as the version in the given commit.

So yes, the same `version` (as part of the `id`) could actually refer to / come from different SHA1s of the source code.

As I *do* like to have the version somewhere in there, what do you think about simply appending the SHA1 to the version in this case, like "0.0.4-3f23fa41632f8935fed028b04335576633ec53a2"?

Reply:
> > I was wondering how the version string arrives at the lockfile.
> 
> I've tested it at the example of `MLHudAlert` by specifying a custom commit in `Podfile` that does not point to a tag and regenerating the lockfile. The version that ends up in `Podfile.lock` is whatever the project's `.podspec` file contains as the version in the given commit.
> 
> So yes, the same `version` (as part of the `id`) could actually refer to / come from different SHA1s of the source code.
> 
> As I _do_ like to have the version somewhere in there, what do you think about simply appending the SHA1 to the version in this case, like "0.0.4-3f23fa41632f8935fed028b04335576633ec53a2"?

This solves indeed the issue I raised. I'm good with that solution.

Reply:
> I'm good with that solution.

Implemented, please have another look.

Reply:
Thanks!